occupi-backend/pkg/router/router.go

```diff
@@ -57,6 +57,8 @@ func OccupiRouter(router *gin.Engine, db *mongo.Client) {
 		auth.POST("/register", middleware.UnProtectedRoute, func(ctx *gin.Context) { handlers.Register(ctx, appsession) })
 		auth.POST("/verify-otp", middleware.UnProtectedRoute, func(ctx *gin.Context) { handlers.VerifyOTP(ctx, appsession) })
 		auth.POST("/logout", middleware.ProtectedRoute, func(ctx *gin.Context) { handlers.Logout(ctx) })
-		// auth.POST("/reset-password", middleware.UnProtectedRoute, func(ctx *gin.Context) { handlers.ForgotPassword(ctx) })
+		// it's typically used by users who can't log in because they've forgotten their password.
+		auth.POST("/forgot-password", func(ctx *gin.Context) { handlers.ResetPassword(ctx, appsession) })
+		auth.POST("/forgot-password-reset", middleware.ProtectedRoute, func(ctx *gin.Context) { handlers.CompletePasswordReset(ctx, appsession)})
```

Comment:
it might make sense to have this route as unprotected as a user who is resetting their password still has not logged on and as such they have no valid session and will be blocked from this route

Reply:
i think we need to have a session where we discuss the different endpoint roles and why some are unprotected, why others dont have any protection etc.

Reply:
okay yes this makes sense, yeah before you add or remove the middleware, just leave it as it is, we'll talk about the logic later this week

Reply:
> i think we need to have a session where we discuss the different endpoint roles and why some are unprotected, why others dont have any protection etc.

okay yes this makes sense, yeah before you add or remove the middleware, just leave it as it is, we'll talk about the logic later this week